bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatAccountHandler.java

```diff
@@ -0,0 +1,449 @@
+    public static String API_ENDPOINT_1 = "https://eurouter.ablecloud.cn:9005/zc-account/v1/";
+    public static String API_ENDPOINT_2 = "https://eurouter.ablecloud.cn:9005/millService/v1/";
+    private static final String ALLOWED_NONCE_CHARACTERS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private static final String REQUEST_TIMEOUT = "300";
+    private final Logger logger = LoggerFactory.getLogger(MillheatAccountHandler.class);
+    private String userId;
+    private String token;
+    private final HttpClient httpClient;
+    private RequestLogger requestLogger;
+    private MillheatDiscoveryService discoveryService;
+    private Gson gson;
+    private MillheatModel model = new MillheatModel(0);
+    private @Nullable ScheduledFuture<?> statusFuture;
+    private MillheatAccountConfiguration config;
+    private Map<ThingUID, ServiceRegistration<DiscoveryService>> discoveryServiceRegistrations = new HashMap<>();
+
+    private static String getRandomString(final int sizeOfRandomString) {
+        final Random random = new Random();
+        final StringBuilder sb = new StringBuilder(sizeOfRandomString);
+        for (int i = 0; i < sizeOfRandomString; ++i) {
+            sb.append(ALLOWED_NONCE_CHARACTERS.charAt(random.nextInt(ALLOWED_NONCE_CHARACTERS.length())));
+        }
+        return sb.toString();
+    }
+
+    public MillheatAccountHandler(Bridge bridge, HttpClient httpClient, BundleContext context) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.httpClient.getContentDecoderFactories().clear();
+        this.httpClient.setUserAgentField(new HttpField("User-Agent", "MillheatApp"));
+        BooleanSerializer serializer = new BooleanSerializer();
+
+        gson = new GsonBuilder().setPrettyPrinting().setDateFormat("yyyy-MM-dd HH:mm:ss")
+                .registerTypeAdapter(Boolean.class, serializer).registerTypeAdapter(boolean.class, serializer).create();
+
+        config = getConfigAs(MillheatAccountConfiguration.class);
+        discoveryService = new MillheatDiscoveryService(this);
+        ServiceRegistration<DiscoveryService> serviceRegistration = context.registerService(DiscoveryService.class,
+                discoveryService, null);
+        discoveryServiceRegistrations.put(this.getThing().getUID(), serviceRegistration);
+        requestLogger = new RequestLogger(bridge.getUID().getId());
+    }
+
+    private boolean allowModelUpdate() {
+        long timeSinceLastUpdate = System.currentTimeMillis() - model.getLastUpdated();
+        if (timeSinceLastUpdate > MIN_TIME_BETWEEEN_MODEL_UPDATES_MS) {
+            return true;
+        }
+        return false;
+    }
+
+    @NonNull
+    public MillheatModel getModel() {
+        return model;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Bridge does not support any commands, but received command " + command + " for channelUID "
```

Comment:
Please use placeholders! I'm sure this is part of the static analysis which we have, please check the report for other things we might have missed.

So please review the report of [the Static code analysis](https://www.openhab.org/docs/developer/guidelines.html#static-code-analysis)

Reply:
Corrected the string.
The static code analysis did not report on this.